openstack/baremetal/v1/nodes/requests.go

```diff
@@ -301,3 +320,118 @@ func Delete(client *gophercloud.ServiceClient, id string) (r DeleteResult) {
 	_, r.Err = client.Delete(deleteURL(client, id), nil)
 	return
 }
+
+// A cleaning step has required keys ‘interface’ and ‘step’, and optional key ‘args’. If specified,
+// the value for ‘args’ is a keyword variable argument dictionary that is passed to the cleaning step
+// method.
+type CleanStep struct {
+	Interface string            `json:"interface,required"`
+	Step      string            `json:"step,required"`
+	Args      map[string]string `json:"args,omitempty"`
+}
+
+// ProvisionStateOptsBuilder allows extensions to add additional parameters to the
+// ChangeProvisionState request.
+type ProvisionStateOptsBuilder interface {
+	ToProvisionStateOpts() (map[string]interface{}, error)
+}
+
+// ProvisionStateOpts for a request to change a node's provision state.
+type ProvisionStateOpts struct {
+	Target         TargetProvisionState `json:"target,required"`
+	ConfigDrive    ConfigDriveBuilder   `json:"-"`
```

Comment:
Why not `omitempty`? This field is only valid for two provisioning actions.

Reply:
Because it’s not being deserialized directly. i wouldn’t review this PR, there’s a lot to change.

Reply:
But for what’s here “-“ is correct. The current iteration of this PR handled encoding, gzipping and making an iso directly. That means it’s not being deserialized through the normal ways, there’s special code to handle config drives.